src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -54,6 +68,14 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    try {
+      List<YoutubePost> newPost = YoutubeApi.getYoutubePost();
+      response.setContentType("text/html;");
+      response.getWriter().println(newPost);
+    } catch (IOException | YoutubeApiException e) {
+      System.out.println("Error: Youtube api returning exception" + e);
+      response.sendError(500, "Server error");
```

Comment:
Since 500 already means it's a server error, I was thinking of a message that is a bit more descriptive. Something like `"An error occurred while fetching posts from YouTube."`?